axelrod/tests/unit/test_hmm.py

```diff
@@ -0,0 +1,135 @@
+        opponent = axelrod.Alternator()
+        for i in range(6):
+            player.play(opponent)
+        self.assertEqual(opponent.history, [C, D] * 3)
+        self.assertEqual(player.history, [D] * 6)
+
+    def test_tft(self):
+        """Tests that the player defined by the table for TFT is in fact
+        TFT."""
+        t_C = [[1, 0], [1, 0]]
+        t_D = [[0, 1], [0, 1]]
+        p = [1, 0]
+        player = axelrod.HMMPlayer(t_C, t_D, p, initial_state=0, initial_action=C)
+        self.assertFalse(player.is_stochastic())
+        self.assertFalse(player.classifier['stochastic'])
+        opponent = axelrod.Alternator()
+        for i in range(6):
+            player.play(opponent)
+        self.assertEqual(opponent.history, [C, D] * 3)
+        self.assertEqual(player.history, [C, C, D, C, D, C])
+
+    def test_wsls(self):
+        """Tests that the player defined by the table for TFT is in fact
+        WSLS (also known as Pavlov."""
+        t_C = [[1, 0], [0, 1]]
+        t_D = [[0, 1], [1, 0]]
+        p = [1, 0]
+        player = axelrod.HMMPlayer(t_C, t_D, p, initial_state=0, initial_action=C)
+        self.assertFalse(player.is_stochastic())
+        self.assertFalse(player.classifier['stochastic'])
+        opponent = axelrod.Alternator()
+        for i in range(6):
+            player.play(opponent)
+        self.assertEqual(opponent.history, [C, D] * 3)
+        self.assertEqual(player.history, [C, C, D, D, C, C])
+
+    def test_malformed_params(self):
+        # Test a malformed table
+        t_C = [[1, 0.5], [0, 1]]
+        self.assertFalse(is_stochastic_matrix(t_C))
+
+        t_C = [[1, 0], [0, 1]]
+        t_D = [[0, 1], [1, 0]]
+        p = [1, 0]
+        hmm = SimpleHMM(t_C, t_C, p, 0)
+        self.assertTrue(hmm.is_well_formed())
+        hmm = SimpleHMM(t_C, t_D, p, -1)
+        self.assertFalse(hmm.is_well_formed())
+        t_C = [[1, -1], [0, 1]]
+        t_D = [[0, 1], [1, 0]]
+        p = [1, 0]
+        hmm = SimpleHMM(t_C, t_D, p, 0)
+        self.assertFalse(hmm.is_well_formed())
+        t_C = [[1, 0], [0, 1]]
+        t_D = [[0, 2], [1, 0]]
+        p = [1, 0]
+        hmm = SimpleHMM(t_C, t_D, p, 0)
+        self.assertFalse(hmm.is_well_formed())
+        t_C = [[1, 0], [0, 1]]
+        t_D = [[0, 1], [1, 0]]
+        p = [-1, 2]
+        hmm = SimpleHMM(t_C, t_D, p, 0)
+        self.assertFalse(hmm.is_well_formed())
+
+
+class TestHMMPlayer(TestPlayer):
+
+    name = "HMM Player"
+    player = axelrod.HMMPlayer
+
+    expected_classifier = {
+        'memory_depth': 1,
+        'stochastic': True,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def test_reset(self):
+        player = self.player([[1]], [[1]], [0], initial_state=0)
+        player.hmm.state = -1
+        player.reset()
+        self.assertFalse(player.hmm.state == -1)
```

Comment:
I think `assertNotEqual` would be better here.

And/or even `assertEqual` with 0.